controllers/provider-aws/pkg/cmd/config.go

```diff
@@ -0,0 +1,140 @@
+// Copyright (c) 2019 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cmd
+
+import (
+	"fmt"
+	"io/ioutil"
+
+	"github.com/gardener/gardener-extensions/controllers/provider-aws/pkg/apis/config"
+	configinstall "github.com/gardener/gardener-extensions/controllers/provider-aws/pkg/apis/config/install"
+	configv1alpha1 "github.com/gardener/gardener-extensions/controllers/provider-aws/pkg/apis/config/v1alpha1"
+
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+
+	"github.com/spf13/pflag"
+)
+
+// ConfigOptions are command line options that can be set for config.ControllerConfiguration.
+type ConfigOptions struct {
+	// Kubeconfig is the path to a kubeconfig.
+	ConfigFilePath string
+
+	config *Config
+}
+
+// Config is a completed controller configuration.
+type Config struct {
+	// Config is the controller configuration.
+	Config *config.ControllerConfiguration
+}
+
+func (c *ConfigOptions) buildConfig(scheme *runtime.Scheme, codecs serializer.CodecFactory) (*config.ControllerConfiguration, error) {
+	if len(c.ConfigFilePath) == 0 {
+		return nil, fmt.Errorf("config file path not set")
+	}
+
+	if err := configinstall.AddToScheme(scheme); err != nil {
+		return nil, err
+	}
+
+	cfg, err := c.loadConfigFromFile(c.ConfigFilePath, codecs)
+	if err != nil {
+		return nil, err
+	}
+
+	cfg, err = c.applyDefaults(cfg, scheme)
+	if err != nil {
+		return nil, err
+	}
+
+	return cfg, nil
+}
+
+func (c *ConfigOptions) loadConfigFromFile(path string, codecs serializer.CodecFactory) (*config.ControllerConfiguration, error) {
+	data, err := ioutil.ReadFile(path)
```

Comment:
As it's done for `clientcmdapi`, could you please provide functions to load configurations into a versioned type?
(See https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/client-go/tools/clientcmd/loader.go#L390)
Then you can use these tested methods here.